services/streammanager/kafka/kafkamanager.go

```diff
@@ -147,6 +149,9 @@ func NewProducerForConfluentCloud(destinationConfig interface{}) (sarama.SyncPro
 
 	var destConfig = ConfluentCloudConfig{}
 	jsonConfig, err := json.Marshal(destinationConfig)
+	if err != nil {
+		pkgLogger.Errorf("Error while marshalling destination config %v", err)
```

Comment:
Lets wrap this error and return it here.

Reply:
done